### PR TITLE
Jarv/newrelic log checks

### PR DESCRIPTION
**Pull request by jarv**
This adds an optional plugin for newrelic to grep for strings in any file on disk and report the occurrences of the match to newrelic.

The extra tasks are only run if `NEWRELIC_LOGWATCH` is defined which is set to a list of templates for watching agents that are checked into the newrelic role template directory.

### CODE REVIEW

File: playbooks/roles/newrelic/templates/opt/newrelic_platform_plugins/newrelic_logwatcher_agent/config/logwatch-certs-errors.j2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% if SANDBOX_USERNAME is defined %}
+{% set server_name = SANDBOX_USERNAME %}
+{% else %}
+{% set server_name = COMMON_ENVIRONMENT|default('unknown-env') + '-' + COMMON_DEPLOYMENT|default('unknown-deployment') %}
```
Comment (feanil) [inline playbooks/roles/newrelic/templates/opt/newrelic_platform_plugins/newrelic_logwatcher_agent/config/logwatch-certs-errors.j2:4]: There are default values for both COMMON_ENVIRONMENT and COMMON_DEPLOYMENT in the `common` role, why not just use those?

**Comment by fredsmith** [comment]
:+1:

File: playbooks/roles/newrelic/templates/opt/newrelic_platform_plugins/newrelic_logwatcher_agent/config/logwatch-certs-errors.j2 (hunk 1 above)
Comment (jarv) [inline playbooks/roles/newrelic/templates/opt/newrelic_platform_plugins/newrelic_logwatcher_agent/config/logwatch-certs-errors.j2:4]: This is a decision on whether we want this to bomb out here or use default values if those vars are not defined. 
The `newrelic` role doesn't depend on common (though it's often included with common) which is why I set defaults if those vars are not defined

**Comment by feanil** [comment]
:+1:

**Pull request merged**